### PR TITLE
shell: documentation update

### PR DESCRIPTION
**Pull request by jakub-uC**
1. Update shell documentation according to proposals in PR #12437 (Extend shell meta keys).
2. Fix lines exceeding 80 characters limit.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

File: doc/subsystems/shell/shell.rst [hunk 1]
```diff
@@ -39,13 +39,15 @@ See the :ref:`shell_api` documentation for more information.
...
-On macOS JLinkRTTClient won't let you enter input. Instead, please use following procedure:
+On macOS JLinkRTTClient won't let you enter input. Instead, please use following
+procedure:
 
 * Open up a first Terminal window and enter:
 
   .. code-block:: none
 
-     JLinkRTTLogger -Device NRF52840_XXAA -RTTChannel 1 -if SWD -Speed 4000 ~/rtt.log
+     JLinkRTTLogger -Device NRF52840_XXAA -RTTChannel 1 -if SWD -Speed 4000
```
Comment (dbkinder) [inline doc/subsystems/shell/shell.rst]: This is a case where the long line is needed (it's all one command line), or you'll need to put the shell line continuation character ``\`` at the end of the first line.  This one is close enough that I'd just join these two line back together.

File: subsys/shell/shell_cmds.c [hunk 2]
```diff
@@ -278,6 +278,13 @@ static int cmd_help(const struct shell *shell, size_t argc, char **argv)
 		" all commands or its subcommands.\n"
 		"You can try to call commands with <-h> or <--help> parameter"
 		" for more information.");
+#if CONFIG_SHELL_METAKEYS
+	shell_print(shell,
+		"Shell supports following meta-keys:\n"
+		"Ctrl+a, Ctrl+b, Ctrl+c, Ctrl+d, Ctrl+e, Ctrl+f, Ctrl+k,"
+		" Ctrl+l, Ctrl+u, Ctrl+w\nAlt+b, Alt+f.\nPlease refere to"
```
Comment (dbkinder) [inline subsys/shell/shell_cmds.c]: s/refere/refer/

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/subsystems/shell/shell.rst (hunk 1 above)
Comment (jarz-nordic) [reply doc/subsystems/shell/shell.rst]: fixed

File: subsys/shell/shell_cmds.c (hunk 2 above)
Comment (jarz-nordic) [reply subsys/shell/shell_cmds.c]: fixed

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks!

**Pull request merged**